File: core/src/ru/myitschool/dcrawler/entities/skills/targeting/FloorSingleDisplayer.java
package ru.myitschool.dcrawler.entities.skills.targeting;

import com.badlogic.gdx.utils.Array;
import ru.myitschool.dcrawler.dungeon.DungeonMap;
import ru.myitschool.dcrawler.entities.skills.Skill;

/**
 * Created by Voyager on 08.03.2018.
 */
public class FloorSingleDisplayer implements TargetDisplayer {

    @Override
    public boolean targetCreation(int x, int y, Array<TilePos> array, Skill skill) {
        if (DungeonMap.getCell(x, y) != null) {
            array.add(new TilePos(x, y, TargetRenderer.getDefaultTargetTile()));
            return true;
        }
        return false;
    }
}

File: core/src/com/wolg_vlad/dcrawler/entities/skills/targeting/FloorSingleDisplayer.kt
package com.wolg_vlad.dcrawler.entities.skills.targeting

import com.badlogic.gdx.utils.Array
import com.wolg_vlad.dcrawler.dungeon.DungeonMap
import com.wolg_vlad.dcrawler.entities.skills.Skill

/**
 * Created by Voyager on 08.03.2018.
 */
class FloorSingleDisplayer : TargetDisplayer {
    override fun targetCreation(x: Int, y: Int, array: MutableList<TilePos>, skill: Skill): Boolean {
        if (DungeonMap.getCell(x, y) != null) {
            array.add(TilePos(x, y, TargetRenderer.defaultTargetTile))
            return true
        }
        return false
    }
}